### ISSUE TITLE
Rubocop LSP queues and reports out-dated diagnostics

### ISSUE THREAD

**Issue by scriabin-enjoyer** [rubocop/rubocop#14612]
- Ubuntu 22.04, x86_64 GNU/Linux
- Using rbenv with global CRuby version 3.3.7
- Rubocop version 1.81.1
- Neovim v0.10: Neovim's LSP client starts the rubocop server with `rubocop --lsp` with no other configuration.
- VSCode 1.105.1
--------
## Problem Description

On October 17, I ran `gem update` to update all of my system-wide gems. This updated rubocop from 1.75.7 to 1.81.1. After this update, the rubocop language server started to emit stale and irrelevant diagnostic messages for text document changes that occurred many keystrokes in the past. It's like rubocop queues up all the incremental changes and starts emitting the various diagnostic messages for each of the changes, not reflecting the current state of the document. It is only after many seconds (sometimes up to 1 minute +) that the diagnostic messages in my editor reflect the true state of the text document. See the video for an example of this behaviour:

https://github.com/user-attachments/assets/8ab21327-87b0-48ad-a012-449615890f55

## Steps to reproduce the problem

This problem may be reproduced using VSCode. I've downloaded a [portable version](https://code.visualstudio.com/download) of VSCode, and installed only the official rubocop extension. I've installed VSCode via the .tar.gz file for linux x64. This is what I am using in the video below

## Expected behavior

I expect the language server to receive the textDocument/didChange notifications from the client, but drop the "irrelevant" changes, and emit diagnostics only for the state of the document that "actually matters". In other words, I expect that the rubocop language server to NOT emit diagnostics for text document states that no longer exist.

## Actual behavior

Here is another video, this time using VSCode, that shows what's happening:

https://github.com/user-attachments/assets/0456d1ae-8ea8-4e72-9e8b-563bbc22a1f8

## RuboCop version

Output of `rubocop -V`
```bash
$ rubocop -V
1.81.1 (using Parser 3.3.9.0, Prism 1.6.0, default, rubocop-ast 1.47.1, analyzing as Ruby 3.3, running on ruby 3.3.7) [x86_64-linux]
```

## Additional Notes
I also have LSP logs from neovim, together with another video that shows the inputs from which the logs were generated. The log file is a bit large to post in a GH issue (I don't use GH all that much, perhaps there is a good way to share big files like this?). In any case, please let me know if it would be helpful to see all the json-rpc messages that are exchanged between neovim and the rubocop server (not sure how to do it with vscode).

**Comment by scriabin-enjoyer** [comment]
FIXED:
Switched to ruby 3.4.0,
ran `gem install rubocop`

everything works fine now lol
